bql/planner/planner.go

```diff
@@ -788,6 +789,204 @@ func (p *queryPlan) String() string {
+					if v.T == nil {
+						return nil, fmt.Errorf("binding %q requires a time, got %+v instead", cc.OAnchorBinding, v)
+					}
+					op, err := predicate.NewTemporal(cc.OID, *v.T)
+					if err != nil {
+						return nil, err
+					}
+					obj = triple.NewPredicateObject(op)
+				}
+			}
+			t, err := triple.New(sbj, prd, obj)
+			if err != nil {
+				return nil, err
+			}
+			if len(cc.ReificationClauses()) > 0 {
+				rts, bn, err := t.Reify()
+				if err != nil {
+					fmt.Errorf("triple.Reify failed to reify %v with error %v", t, err)
+					return nil, err
+				}
+				for _, trpl := range rts[1:] {
+					ts = append(ts, trpl)
+				}
+				for _, rc := range cc.ReificationClauses() {
+					rprd, robj := rc.P, rc.O
+					if rprd == nil {
+						if tbl.HasBinding(rc.PBinding) {
+
+							// Try to bind the predicate.
+							v, ok := r[rc.PBinding]
+							if !ok {
+								return nil, fmt.Errorf("row %+v misses binding %q", r, rc.PBinding)
+							}
+							if v.P == nil {
+								return nil, fmt.Errorf("binding %q requires a predicate, got %+v instead", rc.PBinding, v)
+							}
+							rprd = v.P
+						} else if rc.PTemporal && rc.PAnchorBinding != "" {
+
+							// Try to bind the predicate anchor.
+							v, ok := r[rc.PAnchorBinding]
+							if !ok {
+								return nil, fmt.Errorf("row %+v misses binding %q", r, rc.PAnchorBinding)
+							}
+							if v.T == nil {
+								return nil, fmt.Errorf("binding %q requires a time, got %+v instead", rc.PAnchorBinding, v)
+							}
+							rprd, err = predicate.NewTemporal(rc.PID, *v.T)
+							if err != nil {
+								return nil, err
+							}
+						}
+					}
+					if robj == nil {
+						if tbl.HasBinding(rc.OBinding) {
+
+							// Try to bind the object
+							v, ok := r[rc.OBinding]
+							if !ok {
+								return nil, fmt.Errorf("row %+v misses binding %q", r, rc.OBinding)
+							}
+							co, err := cellToObject(v)
+							if err != nil {
+								return nil, err
+							}
+							robj = co
+						} else if rc.OTemporal && rc.OAnchorBinding != "" {
+
+							// Try to bind the object anchor.
+							v, ok := r[rc.OAnchorBinding]
+							if !ok {
+								return nil, fmt.Errorf("row %+v misses binding %q", r, rc.OAnchorBinding)
+							}
+							if v.T == nil {
+								return nil, fmt.Errorf("binding %q requires a time, got %+v instead", rc.OAnchorBinding, v)
+							}
+							rop, err := predicate.NewTemporal(rc.OID, *v.T)
+							if err != nil {
+								return nil, err
+							}
+							robj = triple.NewPredicateObject(rop)
+						}
+
+					}
+					rt, err := triple.New(bn, rprd, robj)
+					if err != nil {
+						return nil, err
+					}
+					ts = append(ts, rt)
+				}
+
+			} else {
+				ts = append(ts, t)
+			}
+		}
+	}
+	return tbl, update(ctx, ts, p.stm.OutputGraphNames(), p.store, func(g storage.Graph, d []*triple.Triple) error {
+		trace(p.tracer, func() []string {
+			return []string{"Inserting triples to graph \"" + g.ID(ctx) + "\""}
+		})
+		return g.AddTriples(ctx, d)
```

Comment:
This can run on large collections of data. I would suggest that instead of pilling them up on a large slice, you pass them in a channel where you have a goroutine consuming it. See how buffered channels are used elsewhere.  This would also help batch the inserts into the the gorouting doing a few calls to AddTriples instead of a big one. You can make the case that the table can also be big, yes, that is something I have on the todo to revisit :/